cmd/mdatagen/templates/doc.go.tmpl

```diff
@@ -0,0 +1,17 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate mdatagen metadata.yaml
```

Comment:
I don't think it'll work. You cannot generate "generate" pragmas

Reply:
I don't think it would work for the initial creation, but it would for the n+1 times?

Reply:
> I don't think it would work for the initial creation, but it would for the n+1 times?

How does the initial creation happen then? This pragma is currently required for the generation

Reply:
We create the original metadata.yaml file, and from the directory type `mdatagen metadata.yaml`. It works, though is a bit manual.